sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1250,7 +1252,21 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
    */
   override def visitTableName(ctx: TableNameContext): LogicalPlan = withOrigin(ctx) {
     val tableId = visitMultipartIdentifier(ctx.multipartIdentifier)
-    val table = mayApplyAliasPlan(ctx.tableAlias, UnresolvedRelation(tableId))
+    val properties = new util.HashMap[String, String]
+    val timeTravel = if (ctx.temporalClause != null && ctx.temporalClause.version != null) {
+      TimeTravelSpec.create(None, Some(ctx.temporalClause.version.getText))
+    } else if (ctx.temporalClause != null && ctx.temporalClause.timestamp != null) {
+      val tsWithQuotation = ctx.temporalClause.timestamp.getText
+      val ts = tsWithQuotation.substring(1, tsWithQuotation.length - 1)
+      TimeTravelSpec.create(Some(ts), None)
+    } else {
+      None
+    }
+
+    val table = mayApplyAliasPlan(ctx.tableAlias,
+      UnresolvedRelation(tableId,
+        new CaseInsensitiveStringMap(properties),
```

Comment:
do we need to pass this always-empty properties?

Reply:
Sorry, removed.